modules/dhcp/providers/server/isc.rb

```diff
@@ -146,31 +146,34 @@ def loadSubnetData subnet
       initialize_memory_store_with_dhcp_records(parse_config_and_leases_for_records)
     end
 
+    SUBNET_BLOCK_REGEX = /subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)\s*\{([\w-]+\{[^{}]*\}|[\w-][^{}]*;)*\}/
```

Comment:
re comment #1: will fix the space issue; We can't completely ignore what's inside of curly braces, otherwise regex will either match everything between the first and the last curly braces (eager matching) or between the first and the second ones (lazy matching) (I did briefly consider rewriting this bit using a normal parser. I may return to this idea in the future if maintaining these regexes will turn out to be too much pain). Th would work with simpler subnet definitions, but will break with subnet definitions that contain pool and other similar definitions (I believe this is what happened in @brandonweeks setup).

re #2: Yeah, subnet definition can be quite messy; I went over the docs and (I think) covered possible use-cases with tests.